### PR TITLE
Confdev

### PR DESCRIPTION
**Pull request by bhavanaananda**
Conference item changes

### CODE REVIEW

File: app/controllers/conference_item_files_controller.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# -*- encoding : utf-8 -*-
+
+class ConferenceItemFilesController < ApplicationController
+  before_action :set_conferenceItem, only: [:destroy]
```
Comment (anusharanganathan) [inline app/controllers/conference_item_files_controller.rb]: You have named the method is set_conference_item not set_conferenceItem

File: app/controllers/conference_items_controller.rb [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+require 'blacklight_advanced_search'
+
+# bl_advanced_search 1.2.4 is doing unitialized constant on these because we're calling ParseBasicQ directly
+require 'parslet'  
+require 'parsing_nesting/tree'
+
+require "utils"
+
+class ConferenceItemsController < ApplicationController
+  before_action :set_conference_item, only: [:show, :edit, :edit_detailed, :update, :destroy, :revoke_permissions]
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: There is not edit_detailed method

File: app/controllers/conference_items_controller.rb [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+  #   @agreement = DatasetAgreement.new
+  #   @agreement.title = "Agreement for #{@pid}"
+  #   @agreement.agreementType = "Individual"
+  #   @agreement.contributor = current_user.user_key
+  #   @dataset.hasRelatedAgreement = @agreement
+  #   relevant_agreements
+  #   principal_agreement
+  #   @model = 'dataset'
+  # end
+
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: Please delete the lines above

File: app/controllers/conference_items_controller.rb [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+
+  def edit
+    authorize! :edit, params[:id]
+    if @conferenceItem.workflows.first.current_status == "Migrate"
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: This code needs to be similar to the current version (1.3). This is taken from a previous version. See https://github.com/bodleian/oraingest/blob/dev/app/controllers/articles_controller.rb#L93

File: app/controllers/conference_items_controller.rb [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+  #   if @conferenceItem.workflows.first.current_status == "Migrate"
+  #     raise CanCan::AccessDenied.new("Not authorized to edit while record is being migrated!", :read, ConferenceItem)
+  #   end
+  #   authorize! :review, params[:id]
+  #   @pid = params[:id]
+  #   @files = contents
+  #   @model = 'conferenceItem'
+  #   render "edit_detailed"
+  # end
+
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: Please delete the above commented out lines

File: app/controllers/conference_items_controller.rb [hunk 6]
```diff
@@ -0,0 +1,475 @@
...
+    @conferenceItem = ConferenceItem.find_or_create(@pid)
+    @conferenceItem.apply_permissions(current_user)
+
+    if params.has_key?(:files)
+      create_from_upload(params)
+    elsif params.has_key?(:conference_item)
+      add_metadata(params[:conference_item], "")
+    # elsif can? :review, @conferenceItem
+    #     format.html { render action: 'edit_detailed' }
+    #     format.json { render json: @conferenceItem.errors, status: :unprocessable_entity }
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: Please delete the commented out lines

File: app/controllers/conference_items_controller.rb [hunk 7]
```diff
@@ -0,0 +1,475 @@
...
+    if params.has_key?(:redirect_field)
+      redirect_field = params[:redirect_field]
+    end
+    if params.has_key?(:files)
+      create_from_upload(params)
+    elsif conference_item_params
+      add_metadata(params[:conference_item], redirect_field)
+    # elsif can? :review, @conferenceItem
+    #   format.html { render action: 'edit_detailed' }
+    #   format.json { render json: @conferenceItem.errors, status: :unprocessable_entity }
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: Please delete commented out lines

File: app/controllers/conference_items_controller.rb [hunk 8]
```diff
@@ -0,0 +1,475 @@
...
+    else
+      format.html { render action: 'edit' }
+      format.json { render json: @conferenceItem.errors, status: :unprocessable_entity }
+    end
+  end
+
+  def destroy
+    authorize! :destroy, params[:id]
+    if @conferenceItem.workflows.first.current_status == "Migrate"
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: The code here needs to reflect changes from the latest version in production, not the old version. See https://github.com/bodleian/oraingest/blob/dev/app/controllers/articles_controller.rb#L154

File: app/controllers/conference_items_controller.rb [hunk 9]
```diff
@@ -0,0 +1,475 @@
...
+    end
+  end
+
+
+  def add_metadata(conference_item_params, redirect_field)
+    if !@conferenceItem.workflows.nil? && !@conferenceItem.workflows.first.entries.nil?
+      old_status = @conferenceItem.workflows.first.current_status
+    else
+      old_status = nil
+    end
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: The above code will need to change once I have pulled in my changes

File: app/controllers/conference_items_controller.rb [hunk 10]
```diff
@@ -0,0 +1,475 @@
...
+    respond_to do |format|
+      if @conferenceItem.save
+        if can? :review, @conferenceItem
+          format.html { redirect_to edit_conference_item_path(@conferenceItem), notice: 'Conference item was successfully updated.', flash:{ redirect_field: redirect_field } }
+          format.json { head :no_content }
+        else
+          format.html { redirect_to edit_conference_item_path(@conferenceItem), notice: 'Conference item was successfully updated.' }
+          format.json { head :no_content }
+        end
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: The if and else seem to do the same thing - looks like you copied it from articles, where the paths for were different based on access levels. It is not the case here.

File: app/controllers/conference_items_controller.rb [hunk 11]
```diff
@@ -0,0 +1,475 @@
...
+      # elsif can? :review, @conferenceItem
+      #   format.html { render action: 'edit_detailed' }
+      #   format.json { render json: @conferenceItem.errors, status: :unprocessable_entity }
+      else
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: If you have created a new conference object which is only associated to this one conference item, you would need to rollback that change here. See https://github.com/bodleian/oraingest/blob/dev/app/controllers/datasets_controller.rb#L368

File: app/controllers/conference_items_controller.rb [hunk 12]
```diff
@@ -0,0 +1,475 @@
...
+    end
+    #if (@conference.nil? or @conference.agreementType.first == "Individual") and conference_item_params.has_key?(:conference)
+    if @conference.nil?  and conference_item_params.has_key?(:conference)
+      conference_item_params = {}
+      conference_item_params = conference_item_params[:conference]
+      if @conference.nil?
+        @conference = Conference.find_or_create(conf_pid)
+        @conference.apply_permissions(current_user)
+      end
+      conference_item_params[:title] = "Conference for #{@conferenceItem.id}"
```
Comment (anusharanganathan) [inline app/controllers/conference_items_controller.rb]: You should not do this. The user will be adding a title for the conference. This was relevant for dataset agreements, but certainly not here.

**Comment by bhavanaananda** [comment]
The 'confdev' was many commits behind the bodleian/dev. I have merged the bodleian/dev into confdev and commited the code back into 'confdev'. So I've included the newer bits into conference controllers and pulled in the new changes as well.

**Pull request closed without merging**